// Generated by funapi_initiator
// This file is an example to illustrate how to use protobuf
// client-server messages on Funapi.
// You can edit this file to meet your taste.


// You should use a relative path to the system top-level directory.
// On Linux, it's /usr/include.
// To import /usr/include/funapi/network/fun_messages.proto,
// use relative path like this:
import "funapi/network/fun_message.proto";

// Your specific message class.
message LobbyLoginRequest {
  required string id = 1;
  required string type = 2;  // "fb" 외에는 모두 게스트 인증으로 처리
  optional string access_token = 3;  // "fb" 일 경우 클라이언트로부터 받는 토큰 값
}


message LobbyLoginReply {
  required string result = 1;
  optional string msg = 2;
  optional string id = 3;
  optional int32 win_count = 4;
  optional int32 lose_count = 5;
  optional int32 cur_record = 6;
}


message LobbyMatchRequest {
}


message LobbyCancelMatchRequest {
}


message LobbyMatchReply {
  required string result = 1;
  optional string msg = 2;
  optional string player1 = 3;
  optional string player2 = 4;
}


message LobbyRankListRequest {
}


message LobbyRankListReply {

  message RankElement {
    required int64 rank = 1;
    required int64 score = 2;
    required string id = 3;
  }

  required string result = 1;
  optional string msg = 2;
  repeated RankElement rank = 3;
}


message GameStartMessage {
  required string result = 1;
  optional string msg = 2;
}


message GameResultMessage {
  required string result = 1;
  optional string msg = 2;
}


message GameRelayMessage {
  optional float ballX = 1;
  optional float ballY = 2;
  optional float ballVX = 3;
  optional float ballVY = 4;
  optional float barX = 5;
  optional float timeSeq = 6;
}


message PongErrorMessage {
  required string result = 1;
  optional string msg = 2;
}


extend FunMessage {
  optional LobbyLoginRequest lobby_login_req = 20;
  optional LobbyLoginReply lobby_login_repl = 21;

  optional LobbyMatchRequest lobby_match_req = 22;
  optional LobbyCancelMatchRequest lobby_cancel_match_req = 23;
  optional LobbyMatchReply lobby_match_repl = 24;

  optional LobbyRankListRequest lobby_rank_list_req = 25;
  optional LobbyRankListReply lobby_rank_list_repl = 26;

  optional GameStartMessage game_start = 30;
  optional GameResultMessage game_result = 31;
  optional GameRelayMessage game_relay = 32;

  optional PongErrorMessage pong_error = 63;
}


import "funapi/service/multicast_message.proto";

message PbufHelloMessage
{
  optional string message = 1;
}
extend FunMulticastMessage
{
  optional PbufHelloMessage pbuf_hello = 16;
}
